### PR TITLE
Add support for AUTH

### PR DESCRIPTION
**Pull request by makarchuk**
Resolves: #62 
Added support for LOGIN and PLAIN auth methods. Any credentials are accepted.

### CODE REVIEW

File: pkg/server/smtp/handler.go [hunk 1]
```diff
@@ -76,8 +80,14 @@ var commands = map[string]bool{
 	"QUIT":     true,
 	"TURN":     true,
 	"STARTTLS": true,
+	"AUTH":     true,
 }
 
+const (
+	usernameChallenge = "VXNlciBOYW1lAA=="
```
Comment (jhillyerd) [inline pkg/server/smtp/handler.go:87]: Please move these up into the other const block (line 25ish), and add comments explaining these.

**Review by jhillyerd — CHANGES_REQUESTED** [review]
Looks good, one small change

File: pkg/server/smtp/handler.go (hunk 1 above)
Comment (makarchuk) [reply pkg/server/smtp/handler.go]: Moved. Comment Added.

**Comment by jhillyerd** [comment]
Looks great, thank you!

**Comment by jhillyerd** [comment]
Note that I accidently merged this to the wrong branch, have rolled it back from master, but it will go out in the next release candidate, which I am working on this week.

**Pull request merged**